server/pulp/server/controllers/repository.py

```diff
@@ -1027,9 +1027,8 @@ def check_publish(repo_obj, dist_id, dist_inst, transfer_repo, conduit, call_con
     # requested and no predistributor is defined.
     skip_for_predistributor = (predistributor_id and (published_after_predistributor or
                                                       not predistributor_last_published))
-    if skip_for_predistributor or \
-            (last_published and not force_full and not last_updated and not units_removed and
-             not dist_updated and same_override and not predistributor_id):
+    if not force_full and not dist_updated and same_override and (skip_for_predistributor or \
+            (last_published and not last_updated and not units_removed and not predistributor_id)):
```

Comment:
please consider to bring up the `last_published` in the front, by checking last_published first it will be no need to check force_full neither dist_updated in case repo was not published previously at all. 
```
if last_published and not force_full and not dist_updated and same_override and (skip_for_predistributor or \
        (not last_updated and not units_removed and not predistributor_id)):
```